### PR TITLE
Cleanup Code

### LINKED ISSUE #12: Refactor Codebase
Current codebase is a bit out of sync and require some refactoring. Alongwith the middleware Authentication is broken.

PS: The PR for this has already been opened. Opening this to keep progress track

### PR DESCRIPTION
**Pull request by aa0501**
Fixes #12 
Included in this PR:
- Fix Authentication Middleware
- Encapsulate Chats array objects into a separate Message model (would help to reduce Dashboard fetch load and after messages can be load selectively on the basis of which topic is accessed)
- Fix Mongoose schema and Graphql schema sync
- Improve overall structure

Working after this:
- Signup/Login
- Category creation and fetching
- Topic creation and fetching
- Message creation and fetching (per topic)

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,4 +1,3 @@
-MONGO_USER = "codeuino-user"
-MONGO_PASSWORD = "thisisthewrongpassword"
+DATABASE_URL="mongodb://<username>:<password>@cluster0-shard-00-00.t4hho.mongodb.net:27017,cluster0-shard-00-01.t4hho.mongodb.net:27017,cluster0-shard-00-02.t4hho.mongodb.net:27017/<dbname>?ssl=true&replicaSet=atlas-a80apd-shard-0&authSource=admin&retryWrites=true&w=majority"
```
Comment (lazycipher) [inline .env:1]: You might want to remove this URL as it will be different for every mongo cluster.

Comment (aa0501) [reply .env:1]: Yeah! Basically it was just a placeholder to make it clear on what type of URL is expected

**Comment by aa0501** [comment]
@vaibhavdaren @capedcrusader23 @AuraOfDivinity please review and merge if all good

File: .env [hunk 2]
```diff
@@ -1,4 +1,3 @@
...
 JWT_SECRET = "somesupersecretkey"
 PORT=8000
```
Comment (vaibhavdaren) [inline .env:3]:
```suggestion
PORT=8000

```

File: bin/www [hunk 3]
```diff
@@ -50,22 +50,20 @@ function normalizePort(val) {
...
 function onError(error) {
-  if (error.syscall !== 'listen') {
+  if (error.syscall !== "listen") {
     throw error;
   }
 
-  var bind = typeof port === 'string'
-    ? 'Pipe ' + port
-    : 'Port ' + port;
+  var bind = typeof port === "string" ? "Pipe " + port : "Port " + port;
```
Comment (vaibhavdaren) [inline bin/www:57]: check

File: graphql/resolvers/message.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const Message = require("../../models/message");
+const Topic = require("../../models/topic");
+
+module.exports = {
+  messages: async (args, req) => {
+    if (!req.isAuth) {
+      throw new Error("Please login to continue");
```
Comment (vaibhavdaren) [inline graphql/resolvers/message.js:7]: string in constants

**Review by vaibhavdaren — APPROVED** [review]

**Pull request merged**